lib/oauth.js

```diff
@@ -62,10 +65,11 @@ module.exports = new Model({
       this._clientAppId = appID;
 
       // Handle new token details if we've completed a sign in
-      if (checkUrlForToken(window.location.href)) {
+      if (checkUrlForToken(window.location.hash)) {
         console.log('Token found in URL');
-        var tokenDetails = parseUrl(window.location.href);
+        var tokenDetails = parseUrl(window.location.hash);
         this._handleBearerToken(tokenDetails);
+        SESSION_STORAGE.setItem(LOCAL_STORAGE_PREFIX + 'tokenDetails', JSON.stringify(tokenDetails));
 
         // And redirect to the desired page
         var url = ls.get(LOCAL_STORAGE_PREFIX + 'redirectUri');
```

Comment:
Is the redirect straight after this local storage fetch for a saved page location that a user want's to access? E.g. `/#!/transcribe` ? I'm not sure why we redirect again after the authentication server round trip? Perhaps the `redirectUri` name is confusing as it's also used in the oauth authentication flow..

Reply:
I'd assume so, from the comments in the code. I don't know why it requests a new URL from the server, rather than using `pushState` to update the history without reloading the page.

By the way, `location.assign()` can't be passed a relative URL. It must be passed a fully qualified URL. This is what is causing the sign-in bug, because the browser will throw away the bearer token when the page refreshes, and initialise the JS app from scratch.

My best guess as to why the code is written like this would be that this library doesn't know how the client app is set up, so doesn't know how to call a new page short of making a new page request. For example, would a React app need different code here from an Angular app and, if so, why should this library suddenly have to know about javascript app frameworks and how they navigate between views? A better oauth API might do the token initialisation, then leave it up to the calling app to decide how to navigate back to the original view and re-render the page.